NGitLab.Mock/ReleaseCollection.cs

```diff
@@ -40,13 +41,21 @@ public override void Add(ReleaseInfo release)
             base.Add(release);
         }
 
-        public ReleaseInfo Add(string tagName, string name, string description, User user)
+        public ReleaseInfo Add(string tagName, string name, string reference, string description, User user)
         {
             if (tagName is null)
                 throw new ArgumentNullException(nameof(tagName));
             if (user is null)
                 throw new ArgumentNullException(nameof(user));
 
+            if (!Project.Repository.GetTags().Any(r => string.Equals(r.FriendlyName, tagName, StringComparison.Ordinal)))
+            {
+                if (string.IsNullOrEmpty(reference))
+                    throw new GitLabBadRequestException();
```

Comment:
Please add a message indicating why the exception is thrown.

Reply:
done